soh/soh/OTRGlobals.cpp

```diff
@@ -1407,99 +1411,39 @@ extern "C" RandomizerCheck Randomizer_GetCheckFromActor(s16 sceneNum, s16 actorI
     return OTRGlobals::Instance->gRandomizer->GetCheckFromActor(sceneNum, actorId, actorParams);
 }
 
-extern "C" int CopyScrubMessage(u16 scrubTextId, char* buffer, const int maxBufferSize) {
-    std::string scrubText("");
-    int language = CVar_GetS32("gLanguages", 0);
+extern "C" CustomMessageEntry Randomizer_CopyScrubMessage(u16 scrubTextId) {
     int price = 0;
     switch (scrubTextId) {
-        case 0x10A2:
+        case TEXT_SCRUB_POH:
             price = 10;
             break;
-        case 0x10DC:
-        case 0x10DD:
+        case TEXT_SCRUB_STICK_UPGRADE:
+        case TEXT_SCRUB_NUT_UPGRADE:
             price = 40;
             break;
     }
-    switch (language) {
-    case 0: default:
-        scrubText += 0x12; // add the sound
-        scrubText += 0x38; // sound id
-        scrubText += 0x82; // sound id
-        scrubText += "All right! You win! In return for";
-        scrubText += 0x01; // newline
-        scrubText += "sparing me, I will sell you a";
-        scrubText += 0x01; // newline
-        scrubText += 0x05; // change the color
-        scrubText += 0x42; // green
-        scrubText += "mysterious item";
-        scrubText += 0x05; // change the color
-        scrubText += 0x40; // white
-        scrubText += "!";
-        scrubText += 0x01; // newline
-        scrubText += 0x05; // change the color
-        scrubText += 0x41; // red
-        scrubText += std::to_string(price);
-        scrubText += price > 1 ? " Rupees" : " Rupee";
-        scrubText += 0x05; // change the color
-        scrubText += 0x40; // white
-        scrubText += " it is!";
-        scrubText += 0x07; // go to a new message
-        scrubText += 0x10; // message id
-        scrubText += 0xA3; // message id
-            break;
-    case 2:
-        scrubText += 0x12; // add the sound
-        scrubText += 0x38; // sound id
-        scrubText += 0x82; // sound id
-        scrubText += "J'abandonne! Tu veux bien m'acheter";
-        scrubText += 0x01; // newline
-        scrubText += "un ";
-        scrubText += 0x05; // change the color
-        scrubText += 0x42; // green
-        scrubText += "objet myst\x96rieux";
-        //scrubText += ";
-        scrubText += 0x05; // change the color
-        scrubText += 0x40; // white
-        scrubText += "?";
-        scrubText += 0x01; // newline
-        scrubText += "\x84";
-        scrubText += "a fera ";
-        scrubText += 0x05; // change the color
-        scrubText += 0x41; // red
-        scrubText += std::to_string(price) + " Rubis";
-        scrubText += 0x05; // change the color
-        scrubText += 0x40; // white
-        scrubText += "!";
-        scrubText += 0x07; // go to a new message
-        scrubText += 0x10; // message id
-        scrubText += 0xA3; // message id
-        break;
-    }
-
-    return CopyStringToCharBuffer(scrubText, buffer, maxBufferSize);
+    return CustomMessageManager::Instance->RetrieveMessage(Randomizer::scrubMessageTableID, price);
```

Comment:
are we passing in the price as the textId here?

Reply:
Yes, I have it right now so the scrubs have a table to themselves and in scrubsanity, scrubs with the same price can just use the same messageEntry. Definitely subject to change though.

Reply:
it does feel a little weird to me but i makes sense that it works, might be worth making a code cleanup issue to look into ways to make it clearer